templates/terraform/custom_expand/snapshot_encryption_key_expand.go.erb

```diff
@@ -0,0 +1,32 @@
+<%# The license inside this block applies to this file.
+	# Copyright 2018 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData, config *Config) (interface{}, error) {
+	l := v.([]interface{})
+	req := make([]interface{}, 0, 1)
+	if len(l) == 1 && l[0].(map[string]interface{})["raw_key"] != "" {
+		// There is a value
+		outMap := make(map[string]interface{})
+		outMap["rawKey"] = l[0].(map[string]interface{})["raw_key"]
+		req = append(req, outMap)
+	} else {
+		// Check alternative setting?
+		if altV, ok := d.GetOk("snapshot_encryption_key_raw"); ok && altV != "" {
```

Comment:
Can you add a TODO here to remove this as of 2.0.0?

Reply:
Imo no point in a TODO, this is tracked by https://github.com/GoogleCloudPlatform/magic-modules/issues/997 and https://github.com/terraform-providers/terraform-provider-google/issues/1203

Reply:
I think that if this code made it into 2.0.0 it would break the KMS feature. So  something like "TODO remove if found in 2.0" would be a pretty clear visual indicator. But as long as you have a way to ensure that KMS doesn't regress then it doesn't matter what format it's in.